rewrite-gradle/src/test/java/org/openrewrite/gradle/RemoveDependencyTest.java

```diff
@@ -129,6 +129,116 @@ void removeGradleDependencyUsingStringNotationWithExclusion() {
         );
     }
 
+    @Test
+    void removeGradleDependencyWithCommentAfterDependency() {
+        rewriteRun(
+          buildGradle(
+            """
+              plugins {
+                  id 'java-library'
+              }
+              
+              repositories {
+                  mavenCentral()
+              }
+              
+              dependencies {
+                  testImplementation "org.junit.vintage:junit-vintage-engine:5.6.2"        /* comment 1 */ /* comment 2 */ // comment 3
+                  // and more
+                  /* and more */ implementation "org.springframework.boot:spring-boot-starter-web:2.7.0"
+                  def someStatementWithinDepBlock = 33
+                  implementation "com.google.guava:guava:29.0-jre"
+              }
+              """,
+            """
+              plugins {
+                  id 'java-library'
+              }
+              
+              repositories {
+                  mavenCentral()
+              }
+              
+              dependencies {
+                  testImplementation "org.junit.vintage:junit-vintage-engine:5.6.2"        /* comment 1 */ /* comment 2 */ // comment 3
+                  def someStatementWithinDepBlock = 33
```

Comment:
Honestly I don't think we can explain this behavior to the customer nor will it always be correct. I'd be inclined to keep all comments intact and have the user (during a manual review) choose to remove comments that no longer apply.

Reply:
Normally I would agree but in this instance we cannot distinguish between that 5% and 95% as that needs context we do not have.
I agree that test makes sense, but not every customer might unfortunately

Reply:
👍🏻 let's see what other reviewers have to say about this.